packages/transaction-controller/src/helpers/EtherscanRemoteTransactionSource.ts

```diff
@@ -51,20 +56,33 @@ export class EtherscanRemoteTransactionSource
   async fetchTransactions(
     request: RemoteTransactionSourceRequest,
   ): Promise<TransactionMeta[]> {
+    const releaseLock = await this.#mutex.acquire();
+    const acquiredTime = Date.now();
+
     const etherscanRequest: EtherscanTransactionRequest = {
       ...request,
       chainId: request.currentChainId,
     };
 
-    const transactions = this.#isTokenRequestPending
-      ? await this.#fetchTokenTransactions(request, etherscanRequest)
-      : await this.#fetchNormalTransactions(request, etherscanRequest);
+    try {
+      const transactions = this.#isTokenRequestPending
+        ? await this.#fetchTokenTransactions(request, etherscanRequest)
+        : await this.#fetchNormalTransactions(request, etherscanRequest);
 
-    if (this.#includeTokenTransfers) {
-      this.#isTokenRequestPending = !this.#isTokenRequestPending;
-    }
+      if (this.#includeTokenTransfers) {
+        this.#isTokenRequestPending = !this.#isTokenRequestPending;
+      }
 
-    return transactions;
+      return transactions;
+    } finally {
+      const elapsedTime = Date.now() - acquiredTime;
+      const remainingTime = Math.max(0, this.API_FETCH_INTERVAL - elapsedTime);
+      // Wait for the remaining time if it hasn't been 5 seconds yet
+      if (remainingTime > 0) {
+        await new Promise((resolve) => setTimeout(resolve, remainingTime));
+      }
+      releaseLock();
```

Comment:
I feel like we should move this logic into a generic mutex wrapper/helper, but I suppose we can do that if we run into another place where there is rate limiting